### ISSUE TITLE
Should the page really be in the request culture?

### ISSUE THREAD

**Issue by mikeckennedy** [Pylons/pyramid-cookiecutter-starter#15]
Hi guys,

Here come a few pieces of feedback. I'll do them across several issues so we can accept / reject them individually.

At the top of the HTML file in this template, you have

`<html lang="{{ '{{request.locale_name}}' }}"> `

This has been here forever. But I question whether static HTML should really vary in the culture per request. Shouldn't this somehow be either just `en-us` flat or part of the template question?

If neither of these are OK, then let's just not add it. But the worst possible choice seems to vary by request when the content is static.

Thoughts?

**Comment by stevepiercy** [comment]
I think this is needed if the developer wants a multi-language website with localization and without coding templates for each language.

**Comment by mikeckennedy** [comment]
Hi Steve. 

Yes it is needed. But you need it to report actually implemented languages. 

Take this example.

I write a site and localize it into English, Spanish, and German.

A user in China requests the page.  Let's assume I fall back to German if I see a locale I don't support. The text and culture is truly German for the content. But this line:

```<html lang="{{ '{{request.locale_name}}' }}">```

Tells the person's browser it's Chinese, please don't translate, thank you.

Basically, that line assumes we have translated to every conceivable culture value. This is highly unlikely. I would say there is probably no website on the internet that meets this condition.

That's why I think it does more harm than good.

**Comment by mmerickel** [comment]
`request.locale_name` does not return "any" language. It returns the language resolved by the locale negotiator which, by default, is the language defined in the `pyramid.default_locale_name` setting. It is the negotiator's responsibility to only return supported languages and the cookiecutters are generated with `pyramid.default_locale_name = en`.

The only part that's broken right now is if a user abuses the negotiator to set `?_LOCALE_=foo` or sets a cookie named `_LOCALE_=foo` which will trick the **default** negotiator into returning that locale. However that would be weird and the user would be breaking their own browser to do that. In general those values would only be set by the app itself to one of its supported languages.

**Comment by mikeckennedy** [comment]
@mmerickel Thanks for the explanation. I guess I misunderstood about the negotiation. Then it seems OK with me. Sorry for the noise.

**Comment by mmerickel** [comment]
I understand the confusion because on its face it looks like it's a setting coming from the client but indeed it is sanitized by your app's locale negotiator.

**Comment by mikeckennedy** [comment]
Yeah, I closed it. I agree.

Maybe a better name would **have** been `request.negotiated_local` instead of `request.locale_name`. But it's baked in now. :)
